encodings/fastlanes/src/delta/compress.rs

```diff
@@ -3,14 +3,27 @@ use fastlanes::{Delta, Transpose};
 use num_traits::{WrappingAdd, WrappingSub};
 use vortex::array::PrimitiveArray;
 use vortex::compute::unary::fill_forward;
+use vortex::compute::SliceFn;
+use vortex::stats::ArrayStatistics;
 use vortex::validity::Validity;
 use vortex::IntoArrayVariant;
-use vortex_dtype::{match_each_unsigned_integer_ptype, NativePType, Nullability};
+use vortex_dtype::{
+    match_each_signed_integer_ptype, match_each_unsigned_integer_ptype, NativePType, Nullability,
+};
 use vortex_error::VortexResult;
 
 use crate::DeltaArray;
 
 pub fn delta_compress(array: &PrimitiveArray) -> VortexResult<(PrimitiveArray, PrimitiveArray)> {
+    let array = if array.ptype().is_signed_int() {
+        match_each_signed_integer_ptype!(array.ptype(), |$T| {
+            assert!(array.statistics().compute_min::<$T>().map(|x| x >= 0).unwrap_or(false));
+        });
+        &array.reinterpret_cast(array.ptype().to_unsigned())
+    } else {
+        array
+    };
+
```

Comment:
Generally can handle this in the compressor where you would say you will only delta compress unsigned types and then we FoR compressor should kick in to convert things into unsigned and you will only ever see unsinged integers

Reply:
The compressor actually already does this.

Reply:
Done.